language/stdlib/modules/DiemAccount.move

```diff
@@ -1650,27 +1652,38 @@ module DiemAccount {
                 balance_amount >= max_transaction_fee,
                 Errors::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT)
             );
+            // [PCA8]: Check that the gas fee can be paid in this currency
+            assert(
+                TransactionFee::is_coin_initialized<Token>(),
+                Errors::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT)
+            );
         };
 
-        // [PCA8]: Check that the transaction hasn't expired
+        // [PCA9]: Check that the transaction hasn't expired
         assert(
             DiemTimestamp::now_seconds() < txn_expiration_time_seconds,
             Errors::invalid_argument(PROLOGUE_ETRANSACTION_EXPIRED)
         );
 
-        // [PCA9]: Check that the transaction sequence number is not too old (in the past)
+        // [PCA10]: Check that the transaction sequence number is not too old (in the past)
         assert(
             txn_sequence_number >= sender_account.sequence_number,
             Errors::invalid_argument(PROLOGUE_ESEQUENCE_NUMBER_TOO_OLD)
         );
 
-        // [PCA10]: Check that the transaction's sequence number matches the
+        // [PCA11]: Check that the transaction's sequence number matches the
         // current sequence number. Otherwise sequence number is too new by [PCA8].
         assert(
             txn_sequence_number == sender_account.sequence_number,
             Errors::invalid_argument(PROLOGUE_ESEQUENCE_NUMBER_TOO_NEW)
         );
 
+        // [PCA12]: Check that the transaction's sequence number will not overflow.
```

Comment:
It looks like this change missed the "WARNING" comment below. It also missed updating the specification, which I'm working to correct now. I'll reorder this check at the same time.

Reply:
Good spots. 

My understanding is that this ordering is to prevent the "SEQUENCE_NUMBER_TOO_NEW" from clobbering other errors (cf. https://github.com/diem/diem/issues/6321). At the time I made this change my thinking is that this "sequence number overflows a u64" case would be so uncommon as to warrant being at the end, but reordering it to be above this check for consistency is probably the right call (happy to do this as well). Also, cc @runtian-zhou who made this change to comment.

About the specification, I'm not sure if I follow? The specification for the prologue was updated with this change, since otherwise this wouldn't have been able to land since verification of the modules is a land blocking test.

Reply:
See #7063 
I was referring to the Move Adapter specification document. I agree that the specific problem with overflowing the sequence number is unlikely to occur in practice, but it's important to have that "WARNING" comment in the right place so that no other checks are inserted after the "too new" sequence number check.